client/src/main/java/cloud/xline/client/ProtocolClient.java

```diff
@@ -0,0 +1,260 @@
+package cloud.xline.client;
+
+import cloud.xline.client.exceptions.CommandExecutionException;
+import com.curp.protobuf.*;
+import com.curp.protobuf.ProtocolGrpc.ProtocolBlockingStub;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.xline.protobuf.Command;
+import com.xline.protobuf.CommandResponse;
+import com.xline.protobuf.ExecuteError;
+import com.xline.protobuf.SyncResponse;
+import io.grpc.*;
+import org.apache.commons.math3.util.Pair;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.UUID;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.logging.Logger;
+
+public class ProtocolClient {
+
+    private static final Logger logger = Logger.getLogger(ProtocolClient.class.getName());
+    private final State state;
+
+    public ProtocolClient(ArrayList<ManagedChannel> channels) {
+        ArrayList<ProtocolBlockingStub> tmpStubs = new ArrayList<>();
+        for (ManagedChannel channel : channels) {
+            ProtocolBlockingStub stub = ProtocolGrpc.newBlockingStub(channel);
+            tmpStubs.add(stub);
+        }
+        FetchClusterResponse res = this.fetchCluster(tmpStubs);
+        HashMap<Long, ProtocolBlockingStub> stubs = new HashMap<>();
+        long clusterVersion = 0;
+        for (Member member : res.getMembersList()) {
+            // TODO: endpoint load balance
+            ManagedChannel channel = Grpc.newChannelBuilder(member.getAddrs(0), InsecureChannelCredentials.create()).build();
```

Comment:
This could throw an exception.

Reply:
Sorry the description is not clear, I mean the `.getAddrs(0)` could throw an exception if there's no address in the member.

Reply:
`IndexOutOfBoundsException` extends `RuntimeException`, The runtime exception does not need to be declared in the throw statement; it will be thrown automatically.